cmake/check_for_std_filesystem.cmake

```diff
@@ -15,6 +16,8 @@ int main() {
 }
 ]])
 
+set(CMAKE_CXX_STANDARD 17)
+
```

Comment:
Is this code running before we set the variable in the top-level CMakeLists?  Otherwise I can't see how this helps.

Reply:
i have no idea how cmake orders this
the `CMAKE_CXX_STANDARD` is indeed set to C++1z before this module is included